servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java

```diff
@@ -0,0 +1,102 @@
+package io.halkyon.services;
+
+import static java.util.Optional.ofNullable;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.transaction.Transactional;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import io.halkyon.model.Claim;
+import io.halkyon.model.Service;
+import io.halkyon.utils.StringUtils;
+import io.quarkus.scheduler.Scheduled;
+
+/**
+ * The claiming service will poll every new and pending claim to update the claim status according to the following logic:
+ *
+ * A claim can request a service in the form of `<service name>:<service version>`, for example: "mysql:3.6".
+ * If there is a service that matches the criteria of service name, plus service version, then the claim is linked to this service.
+ * Then, if this service has been found in a cluster (the service available flag is true), and there is at least one service
+ * credential, then the claim status will change to "Bind"
+ * this service will
+ * be linked to the claim and the claim status will change to "claimed". Otherwise, the status will be "pending".
+ * After a number of attempts have been made to find a suitable service, the claim status will change to "error".
```

Comment:
This needs to be updated as well

Reply:
there were some left overs here, my bad. it should be fixed now.